src/uct/tcp/tcp_listener.c

```diff
@@ -93,7 +92,12 @@ UCS_CLASS_INIT_FUNC(uct_tcp_listener_t, uct_cm_h cm,
     backlog               = (params->field_mask & UCT_LISTENER_PARAM_FIELD_BACKLOG) ?
                             params->backlog : ucs_socket_max_conn();
 
-    status = ucs_socket_server_init(saddr, socklen, 0, backlog, &self->listen_fd);
+    status = ucs_sockaddr_get_port(saddr, &port);
+    if (status != UCS_OK) {
+        goto err;
+    }
+
+    status = ucs_socket_server_init(saddr, socklen, port, backlog, &self->listen_fd);
```

Comment:
ok, i though we could get rid of 'sockaddr' argument, but it's not possible because in some cases need to listen to a specific address
so better not pass 'port' as separate argument, but use the port number inside sockaddr

Reply:
the port inside saddr is used in case of the tcp_listener but when used from tcp_iface, there is no port inside saddr, we generate a random one inside the ucs_socket_server_init() function.
did you mean to move the random port selection to outside this function?

Reply:
in case of tcp_iface , pass 0 as the port inside sadder, to generate random port